frontend/src/lib/MetricsManager.ts

```diff
@@ -210,4 +213,22 @@ export class MetricsManager {
   private track(evName: string, data: Record<string, unknown>): void {
     analytics.track(evName, data)
   }
+
+  // Use the tracking data injected by S4A if the app is hosted there
+  private getHostTrackingData(): Record<string, unknown> {
+    if (
+      window.location.origin.toLowerCase().endsWith(".streamlit.io") &&
```

Comment:
Is there a more specific prefix for S4A that we want to use? (share.streamlit.io, s4a.streamlit.io)?

I wonder if we'd want to someday e.g. host apps on docs.streamlit.io but not want to include those in metrics; but we can also just cross that bridge when we get there.

Reply:
Will double check the specificity of the prefix..

Regarding the docs.streamlit.io example tho.. if it's just a page that has some user apps in iFrames, it would need to set a value for `window.streamlitTracking` for the embedded user app to track anything.  So by doing nothing, nothing's tracked.  

If I didn't get your example right let me know.